lib/commands/alert.js

```diff
@@ -113,18 +113,21 @@ helpers.getAlert = async function getAlert () {
   }
 
   // determine that the name of the button is what is expected
```

Comment:
why cannot we handle this logic inside WDA? https://github.com/appium/WebDriverAgent/blob/f23062fab0228f2835d4324b2a814aa1649d115a/WebDriverAgentLib/Categories/XCUIApplication%2BFBAlert.m#L16 is the method. Just let me know if there are any other conditions to detect alerts and I'll modify the stuff correspondingly.

Reply:
The WDA logic has _always_ failed for Safari alerts. All the info is in this function, so if you want to translate it over, great!

Reply:
Sure, I'll try to do that today. Would you assist me with testing?

Reply:
also, I'll need the page source output for the cases these alerts are visible, also on iPad

Reply:
I can help with testing. There is also a test suite for Safari alerts within `appium-xcuitest-driver` (https://github.com/appium/appium-xcuitest-driver/blob/master/test/functional/web/safari-alerts-e2e-specs.js). A test for the popup is added in this PR.

Sources...
iPhone: https://gist.github.com/imurchie/421b17339a32e64f965e25c779fdd200
iPad: https://gist.github.com/imurchie/99043004e30f66ff2b1986bc808d0064